src/container/srv_container.c

```diff
@@ -2547,3 +2551,163 @@ ds_cont_oid_fetch_add(uuid_t poh_uuid, uuid_t co_uuid, uuid_t coh_uuid,
 	return rc;
 }
 
+/* Send the RPC from a DAOS server instance to the container service */
+int
+ds_cont_svc_set_prop(uuid_t pool_uuid, uuid_t cont_uuid,
+		     d_rank_list_t *ranks, daos_prop_t *prop)
+{
+	int				rc;
+	struct rsvc_client		client;
+	crt_endpoint_t			ep;
+	struct dss_module_info		*info = dss_get_module_info();
+	crt_rpc_t			*rpc;
+	struct cont_prop_set_in		*in;
+	struct cont_prop_set_out	*out;
+
+	D_DEBUG(DB_MGMT, DF_UUID": Setting container prop\n",
+		DP_UUID(cont_uuid));
+
+	rc = rsvc_client_init(&client, ranks);
+	if (rc != 0)
+		D_GOTO(out, rc);
+
+rechoose:
+	ep.ep_grp = NULL; /* primary group */
+	rc = rsvc_client_choose(&client, &ep);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": cannot find pool service: "DF_RC"\n",
+			DP_UUID(cont_uuid), DP_RC(rc));
+		D_GOTO(out_client, rc);
+	}
+
+	rc = cont_req_create(info->dmi_ctx, &ep, CONT_PROP_SET, &rpc);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to create cont set prop rpc: %d\n",
+			DP_UUID(cont_uuid), rc);
+		D_GOTO(out_client, rc);
+	}
+
+	in = crt_req_get(rpc);
+	uuid_clear(in->cpsi_op.ci_hdl);
+	uuid_clear(in->cpsi_op.ci_pool_hdl);
+	uuid_copy(in->cpsi_op.ci_uuid, cont_uuid);
+	uuid_copy(in->cpsi_pool_uuid, pool_uuid);
+	in->cpsi_prop = prop;
+
+	rc = dss_rpc_send(rpc);
+	out = crt_reply_get(rpc);
+	D_ASSERT(out != NULL);
+
+	rc = rsvc_client_complete_rpc(&client, &ep, rc,
+				      out->cpso_op.co_rc,
```

Comment:
I see in another example in ds_pool_svc_create() that the rc value from dss_rpc_send() call is checked to determine what rc value and hint arguments are provided to this function:
	rc = rsvc_client_complete_rpc(&client, &ep, rc,
				      rc == 0 ? out->pro_op.po_rc : -DER_IO,
				      rc == 0 ? &out->pro_op.po_hint : NULL);

Here, perhaps pass out->cpso_op.co_rc only when rc == 0, and the same for cpso_op.co_hint?

Reply:
Sounds like I can leave this as-is for now.